packages/frontend/src/hooks/useLoggedInUser.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+import { APIError, fetchApi } from '../utils/fetch';
+
+function useLoggedInUser() {
+	return useQuery(
+		['currentUser'],
+		() =>
+			fetchApi({
+				path: '/auth/v1/discord/@me',
+				method: 'get',
+			}),
+		{
+			refetchOnWindowFocus: false,
+			retry: (retries, error: APIError) => {
+				console.log(error);
+
+				return retries < 5 && error.payload?.statusCode !== 401;
```

Comment:
You should actually make sure its an `APIError`, i.e. type it as `unknown`, check `if (!(error instanceof APIError)) return false;`, then do the usual check.